File: backend/Models/Order.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace backend.Models
{
    public enum OrderStatus
    {
        Pending = 1,     // Chờ xác nhận
        Confirmed = 2,   // Đã xác nhận
        Processing = 3,  // Đang xử lý
        Shipping = 4,    // Đang giao hàng
        Delivered = 5,   // Đã giao hàng
        Cancelled = 6,   // Đã hủy
        Returned = 7     // Đã trả hàng
    }

    public enum PaymentStatus
    {
        Pending = 1,     // Chưa thanh toán
        Paid = 2,        // Đã thanh toán
        Failed = 3,      // Thanh toán thất bại
        Refunded = 4     // Đã hoàn tiền
    }

    public class Order
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [StringLength(20)]
        public string OrderNumber { get; set; } = string.Empty;

        [Required]
        public string CustomerId { get; set; } = string.Empty;

        [Required]
        public Guid ShippingAddressId { get; set; }

        [Column(TypeName = "decimal(18,2)")]
        public decimal Subtotal { get; set; }

        [Column(TypeName = "decimal(18,2)")]
        public decimal ShippingFee { get; set; } = 0;

        [Column(TypeName = "decimal(18,2)")]
        public decimal Discount { get; set; } = 0;

        [Column(TypeName = "decimal(18,2)")]
        public decimal Total { get; set; }

        public OrderStatus Status { get; set; } = OrderStatus.Pending;

        public PaymentStatus PaymentStatus { get; set; } = PaymentStatus.Pending;

        [StringLength(500)]
        public string? Notes { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
        public DateTime? ConfirmedAt { get; set; }
        public DateTime? PaidAt { get; set; }
        public DateTime? ShippedAt { get; set; }
        public DateTime? DeliveredAt { get; set; }
        public DateTime? CancelledAt { get; set; }

        [StringLength(500)]
        public string? CancelReason { get; set; }

        // Admin approval fields
        [StringLength(450)] // User ID
        public string? ApprovedBy { get; set; }
        public DateTime? ApprovedAt { get; set; }

        // Shipping fields
        public ShippingProvider ShippingProvider { get; set; } = ShippingProvider.ViettelPost;
        
        [StringLength(100)]
        public string? ShippingCode { get; set; }
        
        public ShippingStatus ShippingStatus { get; set; } = ShippingStatus.PendingPickup;
        
        /// <summary>
        /// Shipping tracking history and status updates from provider (JSON format)
        /// </summary>
        [Column(TypeName = "nvarchar(max)")]
        public string? ShippingHistory { get; set; }

        // Navigation properties
        [ForeignKey("CustomerId")]
        public virtual User Customer { get; set; } = null!;

        [ForeignKey("ShippingAddressId")]
        public virtual UserAddress ShippingAddress { get; set; } = null!;

        public virtual ICollection<OrderItem> Items { get; set; } = new List<OrderItem>();
        
        public virtual ShippingRequest? ShippingRequest { get; set; }
    }
}

File: frontend/src/types/order.ts
export interface OrderItem {
  id: number
  orderId: number
  productId: number
  productName: string
  productSku: string
  productImage?: string
  quantity: number
  unitPrice: number
  totalPrice: number
  customization?: {
    color?: string
    stickers?: Array<{
      id: number
      imageUrl: string
      x: number
      y: number
      scale: number
    }>
    designPreview?: string
  }
}

export interface UserInfo {
  id: number
  email: string
  fullName?: string
  phoneNumber?: string
}

export interface ShippingAddress {
  id: number
  fullName: string
  phoneNumber: string
  address: string
  ward: string
  district: string
  province: string
  isDefault: boolean
}

export type OrderStatus = 
  | 'Pending'      // Chờ xác nhận
  | 'Confirmed'    // Đã xác nhận
  | 'Processing'   // Đang xử lý
  | 'Shipping'     // Đang giao hàng
  | 'Delivered'    // Đã giao hàng
  | 'Cancelled'    // Đã hủy
  | 'Returned'     // Đã trả hàng

export type PaymentStatus = 
  | 'Pending'      // Chưa thanh toán
  | 'Paid'         // Đã thanh toán
  | 'Failed'       // Thanh toán thất bại
  | 'Refunded'     // Đã hoàn tiền

export type PaymentMethod = 
  | 'CashOnDelivery'  // Thanh toán khi nhận hàng (COD)
  | 'BankTransfer'    // Thanh toán chuyển khoản

export interface Order {
  id: number
  orderNumber: string
  customerId: number
  customer: UserInfo
  shippingAddress: ShippingAddress
  items: OrderItem[]
  subtotal: number
  shippingFee: number
  discount: number
  total: number
  status: OrderStatus
  paymentStatus: PaymentStatus
  paymentMethod: PaymentMethod
  notes?: string
  createdAt: string
  updatedAt: string
  confirmedAt?: string
  paidAt?: string
  shippedAt?: string
  deliveredAt?: string
  cancelledAt?: string
  cancelReason?: string
  // Shipping-related fields
  shippingProvider: string
  shippingCode?: string
  shippingStatus?: string
  shippingHistory?: Array<{
    timestamp: string;
    status: string;
    description: string;
    location?: string;
  }>
  // Invoice-related fields
  hasInvoice?: boolean
  invoiceId?: number
  invoiceNumber?: string
  invoiceStatus?: 'generated' | 'sent' | 'error'
}

export interface OrderFilters {
  status?: OrderStatus
  search?: string // Tìm theo order number, customer name, email
  dateFrom?: string
  dateTo?: string
  customerId?: number
}

export interface OrderListResponse {
  orders: Order[]
  total: number
  page: number
  pageSize: number
  totalPages: number
}

export interface UpdateOrderStatusRequest {
  orderId: number
  status: OrderStatus
  notes?: string
}

export interface CreateOrderRequest {
  customerId: string // Backend expects string ID from JWT
  shippingAddressId: string // Backend expects Guid as string
  items: Array<{
    productId: number
    quantity: number
    unitPrice: number
    customization?: {
      color?: string
      stickers?: Array<{
        id: number
        imageUrl: string
        x: number
        y: number
        scale: number
      }>
      designPreview?: string
    }
  }>
  shippingFee?: number
  discount?: number
  notes?: string
  paymentMethod?: PaymentMethod
}

export interface OrderStats {
  totalOrders: number
  pendingOrders: number
  processingOrders: number
  shippingOrders: number
  deliveredOrders: number
  cancelledOrders: number
  totalRevenue: number
  todayOrders: number
}

export interface ApiResponse<T> {
  data?: T
  message?: string
  success: boolean
}